core/src/main/java/org/springframework/security/authorization/AuthorizationManager.java

```diff
@@ -50,8 +50,23 @@ default void verify(Supplier<Authentication> authentication, T object) {
 	 * @param authentication the {@link Supplier} of the {@link Authentication} to check
 	 * @param object the {@link T} object to check
 	 * @return an {@link AuthorizationDecision} or null if no decision could be made
+	 * @deprecated Use {{@link #authorize(Supplier, Object)}} instead
 	 */
+	@Deprecated(forRemoval = true)
 	@Nullable
-	AuthorizationDecision check(Supplier<Authentication> authentication, T object);
+	default AuthorizationDecision check(Supplier<Authentication> authentication, T object) {
+		return (AuthorizationDecision) authorize(authentication, object);
+	}
+
+	/**
+	 * Determines if access should be granted for a specific authentication and object.
+	 * @param authentication the {@link Supplier} of the {@link Authentication} to
+	 * authorize
+	 * @param object the {@link T} object to authorize
+	 * @return an {@link AuthorizationResult} or null if no result could be made
+	 * @since 6.3
+	 */
+	@Nullable
+	AuthorizationResult authorize(Supplier<Authentication> authentication, T object);
```

Comment:
Sorry, I may have misspoken here. We can't introduce a new non-default method to an existing interface as that would not be passive. Will you please switch this so that `authorize` has the default implementation?